convert2rhel/unit_tests/pkghandler_test.py

```diff
@@ -1320,21 +1320,115 @@ def test_fix_default_kernel_with_no_incorrect_kernel(self):
 @pytest.mark.parametrize(
     ("version1", "version2", "expected"),
     (
-        ("123-4.fc35", "123-4.fc35", 0),
-        ("123-5.fc35", "123-4.fc35", 1),
-        ("123-3.fc35", "123-4.fc35", -1),
-        (
-            "4.6~pre16262021g84ef6bd9-3.fc35",
-            "4.6~pre16262021g84ef6bd9-3.fc35",
+        pytest.param(
+            "kernel-core-0:4.18.0-240.10.1.el8_3.i86", "kernel-core-0:4.18.0-240.10.1.el8_3.i86", 0, id="NEVRA"
+        ),
+        pytest.param("kernel-core-0:123-5.fc35", "kernel-core-0:123-4.fc35", 1, id="NEVR"),
+        pytest.param("kernel-core-123-3.fc35.aarch64", "kernel-core-123-4.fc35.aarch64", -1, id="NVRA"),
+        pytest.param("kernel-3.10.0-1160.83.1.0.1.el7", "kernel-3.10.0-1160.83.1.el7", 1, id="NVR"),
+        pytest.param(
+            "kernel-core-0:4.6~pre16262021g84ef6bd9-3.fc35",
+            "kernel-core-0:4.6~pre16262021g84ef6bd9-3.fc35",
             0,
+            id="NEVR",
         ),
-        ("2:8.2.3568-1.fc35", "2:8.2.3568-1.fc35", 0),
+        pytest.param("kernel-core-2:8.2.3568-1.fc35", "kernel-core-2:8.2.3568-1.fc35", 0, id="NEVR"),
+        pytest.param(
+            "1:NetworkManager-1.18.8-2.0.1.el7_9.aarch64", "1:NetworkManager-1.18.8-1.0.1.el7_9.aarch64", 1, id="ENVRA"
+        ),
+        pytest.param("1:NetworkManager-1.18.8-2.0.1.el7_9", "1:NetworkManager-1.18.8-3.0.1.el7_9", -1, id="ENVR"),
+        pytest.param("NetworkManager-1.18.8-2.0.1.el7_9", "1:NetworkManager-2.18.8-3.0.1.el7_9", -1, id="NVR&ENVR"),
+        pytest.param("2:NetworkManager-1.18.8-2.0.1.el7_9", "0:NetworkManager-1.18.8-3.0.1.el7_9", 1, id="ENVR"),
     ),
 )
 def test_compare_package_versions(version1, version2, expected):
     assert pkghandler.compare_package_versions(version1, version2) == expected
 
 
+@pytest.mark.parametrize(
+    ("version1", "version2", "exception_message"),
+    (
+        ("kernel-core-0:390-287.fc36", "kernel-0:390-287.fc36", "kernel-core does not match kernel"),
+        ("kernel-core-0:390-287.fc36.aarch64", "kernel-core-0:391-287.fc36.i86", "aarch64 does not match i86"),
+    ),
+)
+def test_compare_package_versions_warnings(version1, version2, exception_message):
+    with pytest.raises(ValueError, match=exception_message):
+        pkghandler.compare_package_versions(version1, version2)
+
+
+# format: name, epoch, version, release, arch
```

Comment:
I think this message is not necessary anymore since you're using `pytest.param` 
```suggestion
```

Reply:
Noted, I'll remove this comment